packages/toolpad-components/src/DatePicker.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+
+import { TextField } from '@mui/material';
+
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DesktopDatePicker, DesktopDatePickerProps } from '@mui/x-date-pickers/DesktopDatePicker';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { createComponent } from '@mui/toolpad-core';
+import { Dayjs } from 'dayjs';
+
+export interface Props extends DesktopDatePickerProps<string, Dayjs> {
+  format: string;
+  separator: string;
+  fullWidth: boolean;
+  variant: 'outlined' | 'filled' | 'standard';
+  size: 'small' | 'medium';
+  sx: any;
+}
+
+const resolveFormat = (format: string, separator: string) => format.replaceAll(' ', separator);
+
+function DatePicker(props: Props) {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs as any}>
+      <DesktopDatePicker
+        inputFormat={resolveFormat(props.format, props.separator)}
+        {...props}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            fullWidth={props.fullWidth}
+            variant={props.variant}
+            size={props.size}
+            sx={props.sx}
+          />
+        )}
+      />
+    </LocalizationProvider>
+  );
+}
+
+export default createComponent(DatePicker, {
+  argTypes: {
+    value: {
+      typeDef: { type: 'string' },
+      onChangeProp: 'onChange',
+      onChangeHandler: (newValue: Dayjs, { format, separator }: Props) => {
```

Comment:
For uniformity, I prefer to keep the convention of

```tsx
value: T
onChange: (newValue: T) => void
```
I prefer to not ad hoc update the signature of `onChangeHandler` until we have more compelling use-cases to do so.

I think we should standardize on `toISOString()` for the format of `value` and `newValue`, so that the developer doesn't have to consider the formatting when handling this date.

for a benchmark, refer to retool

Reply:
> For uniformity, I prefer to keep the convention of
Do you consider accessing props in a change handler is something we would never need? Even though this seems like a good example where this is needed and makes sense

1. Is there a different way to access props in change handler if I wanted to know which format is selected? 
2. If 1st is a NO then that means that we are getting rid of format prop?

Reply:
1. NO
2. We don't the `onChangeHandler` and implement the logic in the component itself. Similar to how we do `onSelectionChange` in the `DataGrid`

_regarding `onChangeHandler`, this is a remnant from a time when we were still very optimistic we would be able build the components library by just taking all the bare MUI components and wrapping each of them with `createComponent`. In the meantime it has become clear that we needed more customization on most components. Or components that don't exist in MUI at all. And so the `onChangeHandler` has become a bit obsolete. Maybe we should even think about getting rid of it altogether_

Reply:
> We can implement the logic in the component itself, where we have access to the live property values

I had that initially done, however using value of datePicker in other places resulted in raw iso string which seemed wrong to me. But sure we can do that

Reply:
I think we can use the [date-only form](https://tc39.es/ecma262/#sec-date-time-string-format) of ISO8601. This is also unambiguously usable as input for `new Date()`